compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -1357,6 +1357,106 @@ SmallVector<Range> ArgCompareOp::getIterationDomain(OpBuilder &builder) {
   return ranges;
 }
 
+SmallVector<utils::IteratorType> ArgCompareOp::getLoopIteratorTypes() {
+  SmallVector<utils::IteratorType> iteratorTypes(getInputRank(),
+                                                 utils::IteratorType::parallel);
+  iteratorTypes[getDimension()] = utils::IteratorType::reduction;
+  return iteratorTypes;
+}
+
+FailureOr<TilingResult>
+ArgCompareOp::getTiledImplementation(OpBuilder &builder,
+                                     ArrayRef<OpFoldResult> offsets,
+                                     ArrayRef<OpFoldResult> sizes) {
+  Location loc = getLoc();
+  int64_t rank = getInputRank();
+  assert(offsets.size() == static_cast<size_t>(rank) &&
+         "Unexpected offsets size");
+  assert(sizes.size() == static_cast<size_t>(rank) && "Unexpected sizes size");
+
+  SmallVector<Operation *> slices;
+  SmallVector<Value> tiledOperands;
+
+  SmallVector<OpFoldResult> strides(rank, builder.getIndexAttr(1));
+  Operation *inputSlice =
+      getSlice(builder, loc, getInputValue(), offsets, sizes, strides);
+  if (!inputSlice) {
+    return emitOpError("failed to slice input");
+  }
+  tiledOperands.push_back(inputSlice->getResult(0));
+  slices.push_back(inputSlice);
+
+  SmallVector<OpFoldResult> outputOffsets, outputSizes;
+  if (failed(getResultTilePosition(builder, 0, offsets, sizes, outputOffsets,
+                                   outputSizes))) {
+    return emitOpError("failed to compute output tile position");
+  }
+
+  SmallVector<OpFoldResult> outputStrides(outputOffsets.size(),
+                                          builder.getIndexAttr(1));
+  Operation *outputValSlice = getSlice(
+      builder, loc, outputValue(), outputOffsets, outputSizes, outputStrides);
+  if (!outputValSlice) {
+    return emitOpError("failed to slice output value");
+  }
```

Comment:
I know this follows the convention of this file, but I don't think we need a NULL check here as it shouldn't fail. Maybe this could be done as a separate cleanup to the whole file.

```suggestion
  Operation *outputValSlice = getSlice(
      builder, loc, outputValue(), outputOffsets, outputSizes, outputStrides);
```

Reply:
https://github.com/iree-org/iree/blob/d867a94aaf35d551905786253c4912a914312a12/compiler/src/iree/compiler/Dialect/LinalgExt/Utils/Utils.cpp#L89-L105

Just curious — why shouldn’t it fall through? Based on the code above, if a type other than RankedTensorType or MemRefType is provided, it looks like the function could return nullptr.

Reply:
drive-by notification: I think the function returns the nullptr just because the compiler would complain if it doesn't. 

My rule about handling a failure or not (i.e., use assert or ignore the check) is that if it is expected or not. If it is an expected failure and we are able to handle it, let's go with handling it. If not, we either assert or `(void)` the return value.

In the tiling implementation, the input should always be a valid LinalgExt op where the inputs and outputs can only be ranked tensor type or memref type. In this context, the failure handling is not required IMO.

Reply:
Oh, it has `assert false ` there lol.

Reply:
>I think the function returns the nullptr just because the compiler would complain if it doesn't.

I just looked and apparently you don't need a default case. `TypeSwitch` is convertible to `ResultT` and will assert when none of the cases are matched.